specs/commandline/version.md

```diff
@@ -0,0 +1,41 @@
+# notation version
+
+## Description
+
+Use `notation version` to print notation version information.
+
+Upon successful execution, the following information is printed.
+
+```text
+Notation: Notary v2, A tool to sign, store, and verify artifacts.
+
+Version:    <MAJOR.MINOR.PATCH>
+```
+
+## Outline
+
+```text
+Print the notation version information
+
+Usage:
+  notation version [flags]
+
+Flags:
+  -h, --help          Help for version
+```
+
+## Usage
+
+### Print notation version information
+
+```shell
+notation version
+```
+
+An example output:
+
+```text
+Notation: Notary v2, A tool to sign, store, and verify artifacts.
+
+Version:    1.0.0
```

Comment:
We also need to show the following metadata with justifications
- Go version: Different go version has different built-in library implementation, including crypto implementation. Note that the same notation version can be compiled with different golang versions.
- Git Commit & Git tree state: Users / downstream may build by their own. We have to know if they are using the right commit with clean tree state.

Reply:
For instances,

oras downloaded from github release page:
```console
$ oras version
Version:        0.15.1
Go version:     go1.19
Git commit:     36dc01d02c54bf410d56d5e16b597eedbe666ec6
Git tree state: clean
```

oras downloaded from homebrew:
```console
$ oras version
Version:    0.15.1+Homebrew
Go version: go1.19.2
```

Reply:
This comment is valid.